core/bug_api.php

```diff
@@ -1333,9 +1375,15 @@ function bug_move( $p_bug_id, $p_target_project_id ) {
 				# Use target project's default category for moves, since there is no match by name.
 				$t_target_project_category_id = config_get( 'default_category_for_moves', null, null, $p_target_project_id );
 			}
-			bug_set_field( $p_bug_id, 'category_id', $t_target_project_category_id );
+			$t_bugdata->category_id = $t_target_project_category_id;
 		}
 	}
+
+	# Attempt to move disk based attachments to new project file directory.
+	$t_bugdata->add_update_callback( file_move_bug_attachments, array( $t_bugdata->id, $t_bugdata->project_id ) );
```

Comment:
Should we use PHP anonymous functions here to define an anonymous function that calls file_move_bug_attachments and pass in the required parameter, rather than getting a separate function name and array of parameters?  This applies to callback registration and all calls to it.